File: Modules/Module0/Python/bicycle_demo.py
#Imports the Bicycle object
from bicycle import Bicycle

"""
Demonstrates the use of the Bicycle object
"""

#Creates an instance of a Bicycle Object with three arguments
test_bike = Bicycle(9, 12, "orange")

#ACCESSORS
#Prints test_bike's gear, speed, and color using accessor functions
print("test_bike's gear =", test_bike.getgear())
print("test_bike's speed =", test_bike.getspeed())
print("test_bike's color =", test_bike.getcolor())

#MUTATORS

print()
#Sets test_bike's gear to 8 using a mutator
test_bike.setgear(8);

#Sets test_bike's speed to 11
test_bike.setspeed(11)

#Sets test_bike's color to "yellow"
test_bike.setcolor("yellow")

#Prints test_bike's gear, speed, and color again
print("test_bike's gear =", test_bike.getgear())
print("test_bike's speed =", test_bike.getspeed())
print("test_bike's color =", test_bike.getcolor())


#List of Bicycles
bikes = [];

bikes.append(test_bike)
bikes.append(Bicycle(4, 8, "Blue"))
bikes.append(Bicycle(6, 13, "Red"))

print("\nBike List:")
for i in range(0, len(bikes)):
    print("bike ", i, "'s gear = ", bikes[i].getgear(), sep="")
    print("bike ", i, "'s speed = ", bikes[i].getspeed(), sep="")
    print("bike ", i, "'s color = ", bikes[i].getcolor(), sep="")

File: Modules/Module0/CPP/Classes/BicycleDemo.cpp
#include <iostream>
#include "Bicycle.h"

using namespace std;

int main() {

    Bicycle bike1;                                  //Calls default constructor
    Bicycle bike2(4, 15, "Blue");                   //Calls constructor with three arguments (int, int, string)
    Bicycle bike3;                                  //Calls default constructor

    bike3.setSpeed(13);
    bike3.setGear(8);
    bike3.setColor("Red");

    Bicycle bikes[4];                               //Array of Bicycle objects
    bikes[0] = bike1;
    bikes[1] = bike2;
    bikes[2] = bike3;
    bikes[3] = Bicycle(6, 9, "Green");

    for(int i = 0; i < 4; i++) {
        cout << "Bike " << i << " Speed: " << bikes[i].getSpeed() << ", Current Gear: " << bikes[i].getGear() << ", Color: " << bikes[i].getColor() << endl;
    }


}